stl/inc/format

```diff
@@ -2027,6 +2035,11 @@ public:
             && _Cat != _Presentation_type_category::_Floating) {
             _Throw_format_error("Modifier requires an integer presentation type for bool");
         }
+
+        if (_Need_arithmetic_or_pointer_presentation_type && _Cat != _Presentation_type_category::_Integer
+            && _Cat != _Presentation_type_category::_Floating && _Cat != _Presentation_type_category::_Pointer) {
+            _Throw_format_error("Zero modifier requires an arithmetic or pointer presentation type");

```

Comment:
We talked about this at the weekly maintainer meeting and concluded that a comment above the data member that it's currently only set for the zero modifier, plus a comment here that we can say "Zero modifier" in the format_error because the data member is specific to that, would be consistent with the existing code, yet helpful in the future if other modifiers are ever specific to "arith or pointer".